app/src/main/java/ch/epfl/sdp/map/PathsHandler.java

```diff
@@ -210,8 +212,16 @@ private void setLayers() {
         }
     }
 
+    private void setInfectedLayerIfNotEmpty(List<Point> infectedMet, String infectedLayerId, String infectedSourceId) {
+        if (!infectedMet.isEmpty()) {
+            setInfectedPointsLayer(infectedLayerId, infectedSourceId,
+                    infectedMet);
+        }
+    }
+
     private void addInfectedMet(double lat, double lon, Timestamp timestamp, List<Point> infected) {
-        ConcreteDataReceiver concreteDataReceiver = new ConcreteDataReceiver(new GridFirestoreInteractor());
+        ConcreteDataReceiver concreteDataReceiver = new ConcreteDataReceiver(
```

Comment:
Why create a new ConcreteDataReceiver object each time we use this method ?

Reply:
Nice catch, I'm going to make it a field in the class, thanks